types/mocha/index.d.ts

```diff
@@ -400,29 +400,14 @@ declare namespace Mocha {
          *
          * - _Only available when invoked via the mocha CLI._
          */
-        (fn: Func): void;
-
-        /**
-         * [bdd, qunit, tdd] Describe a "hook" to execute the given callback `fn`. The name of the
-         * function is used as the name of the hook.
-         *
-         * - _Only available when invoked via the mocha CLI._
-         */
-        (fn: AsyncFunc): void;
+        (fn: Func | AsyncFunc): void;
```

Comment:
What if we would use a rest parameter with a discriminated union type instead of function overrides as a workaround?

For example:
```typescript
type HookFunction = (...args: [Func|AsyncFunc] | [string, (Func|AsyncFunc)?]) => void;
```